src/app/features/auth/form/form.component.ts

```diff
@@ -0,0 +1,144 @@
+    MatFormFieldModule,
+    MatInputModule,
+    MatIconModule,
+    MatButtonModule,
+    CommonModule,
+    ReactiveFormsModule,
+    FormsModule
+  ],
+  template: `
+    <div class="form">
+      <h2>Zaloguj się</h2>
+      <form [formGroup]="loginForm" (ngSubmit)="login()">
+        <div class="email-container">
+          <mat-form-field appearance="outline">
+            <mat-label>E-mail</mat-label>
+            <input matInput placeholder="E-mail" formControlName="email" required />
+            <mat-error *ngIf="loginForm.get('email')?.hasError('required')">Pole wymagane</mat-error>
+            <mat-error *ngIf="loginForm.get('email')?.hasError('whitespace') && loginForm.value.email">
+              Użyto niedozwolonych znaków</mat-error
+            >
+          </mat-form-field>
+        </div>
+        <div class="password-container">
+          <mat-form-field appearance="outline">
+            <mat-label>Hasło</mat-label>
+            <input
+              matInput
+              placeholder="Hasło"
+              [type]="visible === true ? 'text' : 'password'"
+              formControlName="password"
+              required />
+            <button
+              matSuffix
+              mat-icon-button
+              type="button"
+              (click)="togglePasswordVisibility()"
+              color="primary"
+              [disabled]="!this.loginForm.value.password">
+              <mat-icon>remove_red_eye</mat-icon>
+            </button>
+            <mat-error *ngIf="loginForm.get('password')?.hasError('required')">Pole wymagane</mat-error>
+            <mat-error *ngIf="loginForm.get('password')?.hasError('whitespace') && loginForm.value.password">
+              Użyto niedozwolonych znaków</mat-error
+            >
+          </mat-form-field>
+        </div>
+        <button class="button" mat-raised-button color="primary" [disabled]="loginForm.invalid">Zaloguj się</button>
+      </form>
+    </div>
+  `,
+  styles: [
+    `
+      .form {
+        display: flex;
+        flex-direction: column;
+        justify-content: center;
+        align-items: center;
+      }
+
+      h2 {
+        padding: 0 24px 24px;
+      }
+
+      form {
+        display: flex;
+        flex-direction: column;
+        justify-content: center;
+        gap: 12px;
+      }
+
+      mat-form-field {
+        width: 100%;
+      }
+
+      mat-error {
+        font-size: var(--S_font-size)
+      }
+
+      .button {
+        height: 45px;
+        text-transform: uppercase;  
+      }
+    `,
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class FormComponent {
+  private formBuilder = inject(NonNullableFormBuilder);
+  private store = inject<Store<AuthResponse>>(Store);
+
+  visible = false;
+  loginForm = this.createLoginForm();
+
+  private createLoginForm() {
+    const form = this.formBuilder.group({
+      email: this.formBuilder.control('', [
+        Validators.required,
+        Validators.email,
```

Comment:
natywny walidator email angularowy nie lapie koncowek jak `.com`, nalezy napisac wlasny custom validator. Jeśli nie robimy tego w ramach tego taska, to task do backlogu i komentarz do kodu z numerem taska `TODO`